monkey/infection_monkey/transport/island_api_client_errors.py

```diff
@@ -0,0 +1,30 @@
+class IslandAPIError(Exception):
+    """
+    Raised when something goes wrong when calling the Island API
+    """
+
+    pass
+
+
+class IslandAPITimeoutError(IslandAPIError):
+    """
+    Raised when the API request hits a timeout
+    """
+
+    pass
+
+
+class IslandAPIConnectionError(IslandAPIError):
+    """
+    Raised when the API request can't find/connect to the Island
+    """
+
+    pass
+
+
+class IslandAPIRequestFailedError(IslandAPIError):
+    """
+    Raised when the API request fails(malformed request or an error on the API side)
+    """
+
+    pass
```

Comment:
@ilija-lazoroski this is currently unused. Is there a foreseeable use case for this because of which we're adding this?

Reply:
Yep. It will be used in the upcoming tasks.